### PR TITLE
[New Tutorial] Create a simple Voting dApp on ShimmerEVM

### PR DESCRIPTION
**Pull request by anistark**
# Description of change

[New Tutorial] Create a simple Voting dApp on ShimmerEVM

## Links to any relevant issues

[DevEx Issue #10](https://github.com/iotaledger/guild-devx/issues/10)

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Enhancement

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work

### CODE REVIEW

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+);
+```
+Use the `contractClient` for all fetch methods, such as fetching proposals:
+`await contractClient.getProposals(i)`.
+However, to use state-changing methods, you need a `contractSigner` instance, like:
+```javascript!
+const contractSigner = contractClient.connect(signer);
+```
+
+Now, for instance, to vote, you can assign the action to a button and simply call:
```
Comment (lucas-tortora) [inline tutorials/pages/shimmerevm-dapp-voting.md:292]:
```suggestion
Now you can assign the `contractSigner.vote` action to a button and call:
```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+```javascript!
+let tx = contractSigner.vote(i);
+```
+where `i` is the index of the proposal you want to vote for.
```
Comment (lucas-tortora) [inline tutorials/pages/shimmerevm-dapp-voting.md:297]:
```suggestion

* Where `i` is the index of the proposal you want to vote for.
```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+## Result
```
Comment (lucas-tortora) [inline tutorials/pages/shimmerevm-dapp-voting.md:299]:
```suggestion
## Result

```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+If you followed everything above, it should look something like this:
+![Voting Demo](../static/voting-dapp.jpg "Voting dApp UI")
+
+Feel free to make changes to the design to suit your needs.
```
Comment (lucas-tortora) [inline tutorials/pages/shimmerevm-dapp-voting.md:303]:
```suggestion
```

**Review by lucas-tortora — APPROVED** [review]

**Comment by Dr-Electron** [comment]
First things first:
Can you run `yarn format` to make the action pass? 😆

**Comment by anistark** [comment]
> First things first: Can you run `yarn format` to make the action pass? 😆
> 
> Than you still need to add the tutorial config to the docusaurus config

Done.

File: tutorials/docusaurus.config.js [hunk 5]
```diff
@@ -218,6 +218,25 @@ module.exports = {
...
+        preview: '/shimmerevm-dapp-voting.jpg',
+        route: 'tutorials/shimmerevm-dapp-voting',
+        tags: [
+          'text',
+          'shimmer',
+          'shimmerevm',
+          'solidity',
+          'dapp',
+          'voting',
+          'isc',
```
Comment (Dr-Electron) [inline tutorials/docusaurus.config.js:236]:
```suggestion
          'text',
          'shimmer',
          'solidity',
          'iscp',
```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+}
+```
+
+As you need to check if the voter is allowed to vote, you should add the following:
+
+```solidity!
+require(sender.weight != 0, "Has no right to vote");
+```
+
+You also need to check if the voter has already voted, so you should also add the following conditional :
```
Comment (Dr-Electron) [inline tutorials/pages/shimmerevm-dapp-voting.md:76]:
```suggestion
You also need to check if the voter has already voted, so you should also add the following conditional:
```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+```javascript!
+let tx = contractSigner.vote(i);
+```
+
+- Where `i` is the index of the proposal you want to vote for.
+
+## Result
+
+If you followed everything above, it should look something like this:
+![Voting Demo](../static/voting-dapp.jpg 'Voting dApp UI')
```
Comment (Dr-Electron) [inline tutorials/pages/shimmerevm-dapp-voting.md:327]:
```suggestion
![Voting Demo](/voting-dapp.jpg 'Voting dApp UI')
```

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+## Create a Smart Contracts in Solidity
+
+This tutorial will use a simple Voting Smart Contract, that will be called `Voting.sol`.
+
+#### Create and map the structs
+
+The first thing you need is a `struct` called `Voter`:
+
+```solidity!
```
[the remark this answers is not in the document]
Comment (anistark) [reply tutorials/pages/shimmerevm-dapp-voting.md:33]: It adds line number. Looks better and also useful for reference.

[the remark this answers is not in the document]
Comment (Dr-Electron) [reply tutorials/pages/shimmerevm-dapp-voting.md:33]: Ohh. Didn't know that 🤔 Got it

[the remark this answers is not in the document]
Comment (anistark) [reply tutorials/pages/shimmerevm-dapp-voting.md:33]: [Hackmd](https://hackmd.io/@anistark/shimmerevm-dapp) for reference.
Now that you mention, not sure if it'll work in docusaurus. 🥲

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+```solidity!=
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+/**
+ * @title Voting
+ * @dev Implements voting process along with vote delegation
+ */
+contract Voting {
+
```
Comment (Dr-Electron) [inline tutorials/pages/shimmerevm-dapp-voting.md:128]: Who do you want to target with this tutorial? 
If beginners then I guess you should also explain this part? If not ignore this comment :)

**Review by Dr-Electron — APPROVED** [review]

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
Sorry, didn't mean to approve 😅

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+}
+```
+
+Once you have created the `Voter`, you will need `Proposal` `struct` to nominate who to vote on:
+
+```solidity!
+mapping(address => Voter) public voters;
+```
+
+Once you have created the `Voter`, you will need `Proposal` `struct` to nominate who to vote on:
```
[the remark this answers is not in the document]
Comment (anistark) [reply tutorials/pages/shimmerevm-dapp-voting.md:48]: Yes.

**Comment by Dr-Electron** [comment]
Sir, might I ask you to format again? 😆 ❤️

**Comment by anistark** [comment]
> Sir, might I ask you to format again? 😆 ❤️

done Sir.

File: tutorials/pages/shimmerevm-dapp-voting.md [hunk 11]
```diff
@@ -0,0 +1,328 @@
...
+
+## Requirements
+
+You need the following before you get started:
+
+- **Network**: A network where you will deploy and interact with the smart contract. In this case, the ShimmerEVM.
+- **Wallet**: You will need an interface to connect to the network, verify your identity, and pay the transaction fees. You can use any wallet you choose; this tutorial will use [Metamask](https://metamask.io/), but you can use any wallet you are comfortable with.
+- **Smart Contract**: You will need a smart contract to deploy on-chain to handle all state records on-chain and any changes made to them.
+- **Code Editor or IDE**: Have an IDE or any code editor ready to start working.
+- **Client Library**: This tutorial will use [Ethers](https://docs.ethers.org/v5/). But you can use any other library that you feel more comfortable in.
```
[the remark this answers is not in the document]
Comment (anistark) [reply tutorials/pages/shimmerevm-dapp-voting.md]: I updated it. Removed code editor.

[the remark this answers is not in the document]
Comment (Dr-Electron) [reply tutorials/pages/shimmerevm-dapp-voting.md]: Ohh. Missed that somehow 🤔

**Review by Dr-Electron — APPROVED** [review]
Let's merge

**Pull request merged**